### PR TITLE
MM-45945 - Calls: finish i18n of calls components and messages

### PR DESCRIPTION
**Pull request by cpoile**
#### Summary
- Going through v2 PR comments, next up is i18n of calls components.
- Noticed a small bug when testing, so fixed that.

#### Ticket Link
- https://mattermost.atlassian.net/browse/MM-45945

#### Checklist

- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [x] Includes text changes and localization file updates
- [ ] Have tested against the 5 core themes to ensure consistency between them.

#### Device Information
- Android: 12, Pixel 6
- iOS: 15.3.1, iPhone 7 plus

```release-note
Calls: Added i18n localization support
```

### CODE REVIEW

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx [hunk 1]
```diff
@@ -123,19 +124,28 @@ export const CallsCustomMessage = ({
                     style={style.phoneHangupIcon}
                 />
                 <View style={style.messageText}>
-                    <Text style={style.startedText}>{'Call ended'}</Text>
+                    <Text style={style.startedText}>
+                        {intl.formatMessage({
+                            id: 'mobile.calls_call_ended',
+                            defaultMessage: 'Call ended',
+                        })}
```
Comment (enahum) [inline app/products/calls/components/calls_custom_message/calls_custom_message.tsx:131]: You can use the component FormattedText instead, I think is cleaner

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx [hunk 2]
```diff
@@ -123,19 +124,28 @@ export const CallsCustomMessage = ({
...
+                    </Text>
                     <View style={style.endCallInfo}>
-                        <Text style={style.timeText}>{'Ended at '}</Text>
+                        <Text style={style.timeText}>{`${endedAt} `}</Text>
```
Comment (enahum) [inline app/products/calls/components/calls_custom_message/calls_custom_message.tsx:134]: Same here and all others

File: app/products/calls/components/join_call_banner/join_call_banner.tsx [hunk 3]
```diff
@@ -94,7 +94,9 @@ const JoinCallBanner = ({
                 size={16}
                 style={style.joinCallIcon}
             />
-            <Text style={style.joinCall}>{'Join Call'}</Text>
+            <Text style={style.joinCall}>
```
Comment (enahum) [inline app/products/calls/components/join_call_banner/join_call_banner.tsx:97]: Here too

File: app/products/calls/screens/call_screen/call_screen.tsx [hunk 4]
```diff
@@ -377,7 +377,10 @@ const CallScreen = ({componentId, currentCall, participantsDict, teammateNameDis
                     style={style.screenShareImage}
                 />
                 <Text style={style.screenShareText}>
-                    {`You are viewing ${displayUsername(participantsDict[currentCall.screenOn].userModel, teammateNameDisplay)}'s screen`}
+                    {intl.formatMessage({
```
Comment (enahum) [inline app/products/calls/screens/call_screen/call_screen.tsx:380]: Same on this file

**Review by enahum — APPROVED** [review]
Approving but I would like to see the use of FormattedText instead of Text

**Review by streamer45 — APPROVED** [review]
Looks good :+1: As mentioned before let's sync with @justinegeffen on newly added strings.

File: app/products/calls/components/calls_custom_message/calls_custom_message.tsx (hunk 1 above)
Comment (cpoile) [reply app/products/calls/components/calls_custom_message/calls_custom_message.tsx]: Ah, thanks for pointing that out. It is cleaner, thanks.

File: assets/base/i18n/en.json [hunk 5]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_not_available_title": "Calls is not enabled",
   "mobile.calls_ok": "OK",
+  "mobile.calls_raise_hand": "Raise hand",
   "mobile.calls_see_logs": "see server logs",
+  "mobile.calls_speaker": "Speaker",
   "mobile.calls_start_call": "Start Call",
+  "mobile.calls_unmute": "Unmute",
+  "mobile.calls_viewing_screen": "You are viewing {name}'s screen",
+  "mobile.calls_you": "(you)",
   "mobile.camera_photo_permission_denied_description": "Take photos and upload them to your server or save them to your device. Open Settings to grant {applicationName} Read and Write access to your camera.",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:394]:
```suggestion
  "mobile.camera_photo_permission_denied_description": "Take photos and upload them to your server or save them to your device. Open Settings to grant {applicationName} read and write access to your camera.",
```

File: assets/base/i18n/en.json [hunk 6]
```diff
@@ -348,34 +347,50 @@
...
+  "mobile.calls_lasted": "Lasted {duration}",
+  "mobile.calls_leave": "Leave",
   "mobile.calls_leave_call": "Leave Call",
+  "mobile.calls_lower_hand": "Lower hand",
+  "mobile.calls_more": "More",
+  "mobile.calls_mute": "Mute",
+  "mobile.calls_name_is_talking": "{name} is talking",
+  "mobile.calls_name_started_call": "{name} started a call",
+  "mobile.calls_noone_talking": "No one is talking",
   "mobile.calls_not_available_msg": "Please contact your system administrator to enable the feature.",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:383]:
```suggestion
  "mobile.calls_not_available_msg": "Please contact your System Admin to enable the feature.",
```

File: assets/base/i18n/en.json [hunk 7]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_end_permission_msg": "You do not have permission to end the call. Please ask the call creator to end the call.",
   "mobile.calls_end_permission_title": "Error",
+  "mobile.calls_ended_at": "Ended at",
   "mobile.calls_error_message": "Error: {error}",
   "mobile.calls_error_permissions": "no permissions to microphone, unable to start call",
   "mobile.calls_error_title": "Error",
   "mobile.calls_join_call": "Join Call",
+  "mobile.calls_lasted": "Lasted {duration}",
+  "mobile.calls_leave": "Leave",
   "mobile.calls_leave_call": "Leave Call",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:376]:
```suggestion
  "mobile.calls_leave_call": "Leave call",
```

File: assets/base/i18n/en.json [hunk 8]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_end_msg_channel": "Are you sure you want to end a call with {numParticipants} participants in {displayName}?",
+  "mobile.calls_end_msg_channel_default": "Are you sure you want to end the call?",
   "mobile.calls_end_msg_dm": "Are you sure you want to end a call with {displayName}?",
   "mobile.calls_end_permission_msg": "You do not have permission to end the call. Please ask the call creator to end the call.",
   "mobile.calls_end_permission_title": "Error",
+  "mobile.calls_ended_at": "Ended at",
   "mobile.calls_error_message": "Error: {error}",
   "mobile.calls_error_permissions": "no permissions to microphone, unable to start call",
   "mobile.calls_error_title": "Error",
   "mobile.calls_join_call": "Join Call",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:373]:
```suggestion
  "mobile.calls_join_call": "Join call",
```

File: assets/base/i18n/en.json [hunk 9]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_enable": "Enable Calls",
   "mobile.calls_end_call_title": "End call",
   "mobile.calls_end_msg_channel": "Are you sure you want to end a call with {numParticipants} participants in {displayName}?",
+  "mobile.calls_end_msg_channel_default": "Are you sure you want to end the call?",
   "mobile.calls_end_msg_dm": "Are you sure you want to end a call with {displayName}?",
   "mobile.calls_end_permission_msg": "You do not have permission to end the call. Please ask the call creator to end the call.",
   "mobile.calls_end_permission_title": "Error",
+  "mobile.calls_ended_at": "Ended at",
   "mobile.calls_error_message": "Error: {error}",
   "mobile.calls_error_permissions": "no permissions to microphone, unable to start call",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:371]:
```suggestion
  "mobile.calls_error_permissions": "No permissions to microphone, unable to start call",
```

File: assets/base/i18n/en.json [hunk 10]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_call_screen": "Call",
+  "mobile.calls_chat_thread": "Chat thread",
+  "mobile.calls_current_call": "Current call",
   "mobile.calls_disable": "Disable Calls",
   "mobile.calls_enable": "Enable Calls",
   "mobile.calls_end_call_title": "End call",
   "mobile.calls_end_msg_channel": "Are you sure you want to end a call with {numParticipants} participants in {displayName}?",
+  "mobile.calls_end_msg_channel_default": "Are you sure you want to end the call?",
   "mobile.calls_end_msg_dm": "Are you sure you want to end a call with {displayName}?",
   "mobile.calls_end_permission_msg": "You do not have permission to end the call. Please ask the call creator to end the call.",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:367]:
```suggestion
  "mobile.calls_end_permission_msg": "You don't have permission to end the call. Please ask the call creator to end the call.",
```

File: assets/base/i18n/en.json [hunk 11]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.add_team.create_team": "Create a new team",
   "mobile.add_team.join_team": "Join Another Team",
   "mobile.android.back_handler_exit": "Press back again to exit",
   "mobile.android.photos_permission_denied_description": "Upload photos to your server or save them to your device. Open Settings to grant {applicationName} Read and Write access to your photo library.",
   "mobile.android.photos_permission_denied_title": "{applicationName} would like to access your photos",
+  "mobile.calls_call_ended": "Call ended",
   "mobile.calls_call_screen": "Call",
+  "mobile.calls_chat_thread": "Chat thread",
+  "mobile.calls_current_call": "Current call",
   "mobile.calls_disable": "Disable Calls",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:361]:
```suggestion
  "mobile.calls_disable": "Disable calls",
```

File: assets/base/i18n/en.json [hunk 12]
```diff
@@ -348,34 +347,50 @@
...
   "mobile.calls_enable": "Enable Calls",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:362]:
```suggestion
  "mobile.calls_enable": "Enable calls",
```

File: assets/base/i18n/en.json [hunk 13]
```diff
@@ -348,34 +347,50 @@
...
   "mentions.empty.title": "No Mentions yet",
   "mobile.about.appVersion": "App Version: {version} (Build {number})",
-  "mobile.about.serverVersion": "Server Version: {version} (Build {number})",
-  "mobile.about.serverVersionNoBuild": "Server Version: {version}",
   "mobile.account.settings.save": "Save",
   "mobile.action_menu.select": "Select an option",
   "mobile.add_team.create_team": "Create a new team",
   "mobile.add_team.join_team": "Join Another Team",
   "mobile.android.back_handler_exit": "Press back again to exit",
   "mobile.android.photos_permission_denied_description": "Upload photos to your server or save them to your device. Open Settings to grant {applicationName} Read and Write access to your photo library.",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:355]:
```suggestion
  "mobile.android.photos_permission_denied_description": "Upload photos to your server or save them to your device. Open Settings to grant {applicationName} read and write access to your photo library.",
```

File: assets/base/i18n/en.json [hunk 14]
```diff
@@ -348,34 +347,50 @@
   "mentions.empty.paragraph": "You'll see messages here when someone mentions you or uses terms you're monitoring.",
   "mentions.empty.title": "No Mentions yet",
   "mobile.about.appVersion": "App Version: {version} (Build {number})",
-  "mobile.about.serverVersion": "Server Version: {version} (Build {number})",
-  "mobile.about.serverVersionNoBuild": "Server Version: {version}",
   "mobile.account.settings.save": "Save",
   "mobile.action_menu.select": "Select an option",
   "mobile.add_team.create_team": "Create a new team",
   "mobile.add_team.join_team": "Join Another Team",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:353]:
```suggestion
  "mobile.add_team.join_team": "Join another team",
```

File: assets/base/i18n/en.json [hunk 15]
```diff
@@ -348,34 +347,50 @@
   "mentions.empty.paragraph": "You'll see messages here when someone mentions you or uses terms you're monitoring.",
   "mentions.empty.title": "No Mentions yet",
   "mobile.about.appVersion": "App Version: {version} (Build {number})",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:349]:
```suggestion
  "mobile.about.appVersion": "App version: {version} (Build {number})",
```

File: assets/base/i18n/en.json [hunk 16]
```diff
@@ -348,34 +347,50 @@
   "mentions.empty.paragraph": "You'll see messages here when someone mentions you or uses terms you're monitoring.",
   "mentions.empty.title": "No Mentions yet",
```
Comment (justinegeffen) [inline assets/base/i18n/en.json:348]:
```suggestion
  "mentions.empty.title": "No mentions yet",
```

**Review by justinegeffen — APPROVED** [review]
Thanks! I added some changes to the strings. LMK if you need a second pass, otherwise approving for expediency. :)

File: assets/base/i18n/en.json (hunk 14 above)
Comment (enahum) [reply assets/base/i18n/en.json:353]: For anything not related to calls on this PR, plz discuss with UX first then file a ticket

File: assets/base/i18n/en.json (hunk 5 above)
Comment (enahum) [reply assets/base/i18n/en.json:394]: All changes must happen in both, this file and in the code. So apply the suggestions being mundful of that

**Pull request merged**